src/RA_Core.cpp

```diff
@@ -401,8 +401,10 @@ API int CCONV _RA_OnLoadNewRom(const BYTE* pROM, unsigned int nROMSize)
         {
             //	Some other fatal error... panic?
             ASSERT(!"Unknown error from requestgameid.php");
-
-            MessageBox(g_RAMainWnd, NativeStr("Error from " RA_HOST_URL "!\n").c_str(), TEXT("Error returned!"), MB_OK);
+            tstring tstr{TEXT("Error from ")};
+            tstr.append(RA_HOST_URL);
+            tstr.append(TEXT("!\n"));
```

Comment:
Previously, a single string was created at compile time. Unless the compiler optimizes this back into a single string, you're looking at possibly introducing one or more reallocs.

Reply:
I think I remember something about the compiler optimizing the `+ operator`, but I don't remember anything about it optimizing the `append` method.

Reply:
std::string is not a literal type and cannot result in a constant expression. I can see what you mean but the speed is the same. A literal type either has all it's members as literal types or it's default constructor is constexpr.

Reply:
Admittedly, _this_ code probably isn't called frequently enough to worry about execution speed, but it's something I'm always thinking about.

In this case, `NativeStr` is just used to Narrow/Widen the string, it probably should just be a `TEXT`:
```
MessageBox(g_RAMainWnd, TEXT("Error from " RA_HOST_URL "!\n"), TEXT("Error returned!"), MB_OK);
```
The following code results in a single contiguous block of memory at compile tile:
```
const char* text = "Hello" ", " "world";
```
I'm probably remembering another language, since `operator +` isn't available for `const char*`, but I thought this did too:
```
std::string text = "Hello" + ", " + "world";
```

Reply:
That's a linear logarithmic operation. I would need to double check but I don't think variables can be used like that (macros can be though).

Reply:
The way this is done now is guaranteed to built in compile-time and checks that RA_HOST_URL exists in that string_view (readonly string) at compile time (only string literals can be constant expressions, basic_string cannot).